filters/auth/jwt_validation_test.go

```diff
@@ -0,0 +1,210 @@
+	defer backend.Close()
+
+	authServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(489)
+			return
+		}
+
+		allKeys := map[string][]interface{}{}
+		allKeys["keys"] = append(allKeys["keys"], map[string]interface{}{"kid": kid, "n": base64.RawURLEncoding.EncodeToString(privateKey.PublicKey.N.Bytes()), "e": "AQAB"})
+		if r.URL.Path != testAuthPath {
+			w.WriteHeader(488)
+			return
+		}
+
+		e := json.NewEncoder(w)
+		err2 := e.Encode(allKeys)
+		if err2 != nil && err2 != io.EOF {
+			t.Errorf("Failed to json encode: %v", err2)
+		}
+	}))
+	defer authServer.Close()
+
+	testOidcConfig := getTestOidcConfig()
+	issuerServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != TokenIntrospectionConfigPath {
+			w.WriteHeader(486)
+			return
+		}
+		e := json.NewEncoder(w)
+		err := e.Encode(testOidcConfig)
+		if err != nil {
+			t.Fatalf("Could not encode testOidcConfig: %v", err)
+		}
+	}))
+	defer issuerServer.Close()
+
+	// patch openIDConfig to the current testservers
+	testOidcConfig.Issuer = "http://" + issuerServer.Listener.Addr().String()
+	testOidcConfig.JwksURI = "http://" + authServer.Listener.Addr().String() + testAuthPath
+
+	for _, ti := range []struct {
+		msg         string
+		authType    string
+		authBaseURL string
+		args        []interface{}
+		hasAuth     bool
+		auth        string
+		expected    int
+	}{{
+		msg:         "jwtValidation: invalid token",
+		authBaseURL: testAuthPath,
+		args:        []interface{}{},
+		hasAuth:     true,
+		auth:        "invalid-token",
+		expected:    http.StatusUnauthorized,
+	}, {
+		msg:         "jwtValidation: valid token",
+		authBaseURL: testAuthPath,
+		args:        []interface{}{},
+		hasAuth:     true,
+		auth:        createToken(),
+		expected:    http.StatusOK,
+	}} {
+		t.Run(ti.msg, func(t *testing.T) {
+			if ti.msg == "" {
+				t.Fatalf("unknown ti: %+v", ti)
+			}
+
+			var spec = NewJwtValidation(testAuthTimeout)
+
+			args := []interface{}{testOidcConfig.Issuer}
+			args = append(args, ti.args...)
+			f, err := spec.CreateFilter(args)
+			if err != nil {
+				if ti.expected == invalidFilterExpected {
+					return
+				}
+				t.Errorf("error in creating filter for %s: %v", ti.msg, err)
+				return
+			}
+
+			f2 := f.(*jwtValidationFilter)
+			defer f2.Close()
+
+			fr := make(filters.Registry)
+			fr.Register(spec)
+			r := &eskip.Route{Filters: []*eskip.Filter{{Name: spec.Name(), Args: args}}, Backend: backend.URL}
+
+			proxy := proxytest.New(fr, r)
+			defer proxy.Close()
+
+			reqURL, err := url.Parse(proxy.URL)
+			if err != nil {
+				t.Errorf("Failed to parse url %s: %v", proxy.URL, err)
+				return
+			}
+
+			req, err := http.NewRequest("GET", reqURL.String(), nil)
+			if err != nil {
+				t.Errorf("failed to create request %v", err)
+				return
+			}
+
+			if ti.hasAuth {
```

Comment:
Could just check `ti.auth != ""` so there is no need to have `ti.hasAuth` at all